src/common/snippets/src/pass/loop_fusion.cpp

```diff
@@ -0,0 +1,331 @@
+            }
+        }
+    }
+
+    return is_current_reduce_max_case;
+}
+
+} // namespace
+
+
+bool ngraph::snippets::pass::LoopFusion::Merge(const std::shared_ptr<op::LoopBegin>& loop_begin_down) {
+    if (!loop_begin_down) {
+        return false;
+    }
+
+    std::shared_ptr<ngraph::snippets::op::LoopEnd> loop_end_up = nullptr;
+    std::shared_ptr<ngraph::snippets::op::Buffer> buffer = nullptr;
+    // Initialize the corresponding upper LoopEnd and Buffer
+    if (!get_buffer_and_loop_end(loop_begin_down, loop_end_up, buffer)) {
+        return false;
+    }
+    // Check for conditions of fusion
+    if (!can_be_merged(loop_end_up, loop_begin_down)) {
+        return false;
+    }
+
+    const auto loop_end_down = loop_begin_down->get_loop_end();
+    const auto loop_begin_up = loop_end_up->get_loop_begin();
+    const auto new_input_count = loop_begin_up->get_input_size() + loop_begin_down->get_input_size();
+    const auto new_output_count = loop_end_up->get_output_size() + loop_end_down->get_output_size();
+    const auto new_io_count = new_input_count + new_output_count;
+    const auto ptr_increments_up = loop_end_up->get_ptr_increments();
+    const auto ptr_increments_down = loop_end_down->get_ptr_increments();
+    const auto finalization_offsets_up = loop_end_up->get_finalization_offsets();
+    const auto finalization_offsets_down = loop_end_down->get_finalization_offsets();
+    std::vector<int64_t> new_ptr_increments, new_finalization_offsets;
+    new_ptr_increments.reserve(new_io_count);
+    new_finalization_offsets.reserve(new_io_count);
+
+    // Collect new loop inputs
+    std::vector<Edge> loop_inputs;
+    loop_inputs.reserve(new_input_count);
+    new_ptr_increments.reserve(new_io_count);
+    new_finalization_offsets.reserve(new_io_count);
+    collect_loop_inputs(loop_begin_up, buffer, loop_inputs, new_ptr_increments, new_finalization_offsets);
+    collect_loop_inputs(loop_begin_down, buffer, loop_inputs, new_ptr_increments, new_finalization_offsets);
+
+    // Collect new Loop outputs
+    std::vector<Edge> loop_outputs;
+    loop_outputs.reserve(new_output_count);
+    // We can fuse Loop with maximum accumulator pattern only with Smth input
+    // So firstly, we analyze LoopEnd down (it's possible maximum accumulator pattern), set `reduce_max_case` variable
+    // if it's really maximum accumulator pattern, and then analyze LoopEnd up using `reduce_max_case` variable
+    const bool reduce_max_case = collect_loop_outputs(loop_end_down, buffer, loop_outputs, new_ptr_increments, new_finalization_offsets, false);
+    collect_loop_outputs(loop_end_up, buffer, loop_outputs, new_ptr_increments, new_finalization_offsets, reduce_max_case);
+    if (reduce_max_case) {
+        const auto target_inputs = loop_begin_down->output(0).get_target_inputs();
+        NGRAPH_CHECK(target_inputs.size(), "LoopBegin in ReduceMax should have only one consumer (Load) for out port 0");
```

Comment:
Should it be `target_inputs.size() == 1`? Otherwise it won't distinguish between any nonzero consumer numbers

Reply:
Sure! Thanks